mountpoint-s3/src/inode.rs

```diff
@@ -529,6 +417,168 @@ impl SuperblockInner {
             .ok_or(InodeError::InodeDoesNotExist(ino))
     }
 
+    /// Increase the lookup count of the given inode and
+    /// ensure it is registered with this superblock.
+    pub fn remember(&self, inode: &Inode) -> u64 {
+        let lookup_count = inode.inc_lookup_count();
+        if lookup_count == 1 {
+            let previous = self.inodes.write().unwrap().insert(inode.ino(), inode.clone());
+            assert!(previous.is_none(), "inode numbers are never reused");
+        }
+        lookup_count
+    }
+
+    /// Lookup an inode in the parent directory with the given name.
```

Comment:
Worth a comment that the caller is responsible for calling `remember` if the inode is expected to survive beyond the lookup.